executor/join.go

```diff
@@ -529,6 +558,36 @@ func (e *NestedLoopApplyExec) prepare(goCtx goctx.Context) error {
 	}
 }
 
+// prepare4Chunk reads all data from the small table and stores them in a slice.
+func (e *NestedLoopApplyExec) prepare4Chunk(goCtx goctx.Context) error {
+	err := e.SmallExec.Open(goctx.TODO())
+	if err != nil {
+		return errors.Trace(err)
+	}
+	defer terror.Call(e.SmallExec.Close)
```

Comment:
move this line up, we should guarantee `e.SmallExec.Close` is called even if error detected after calling `e.SmallExec.Open`